src/commands/goToDefinition.ts

```diff
@@ -0,0 +1,212 @@
+  }
+
+  if (!jupyter.isActive) {
+    vscode.window.showErrorMessage("Jupyter extension is not active.");
+    return;
+  }
+
+  const activeEditor = vscode.window.activeNotebookEditor;
+  if (!activeEditor) {
+    vscode.window.showInformationMessage("No active Jupyter editor found.");
+    return;
+  }
+
+  if (idx + 1 < activeEditor.notebook.cellCount) {
+    activeEditor.selections = [new vscode.NotebookRange(idx, idx + 1)];
+    activeEditor.revealRange(
+      new vscode.NotebookRange(idx, idx + 1),
+      vscode.NotebookEditorRevealType.AtTop
+    );
+  }
+}
+
+// Tries to find the `export` comment in the cell
+// If found, returns the value, otherwise returns an empty string
+function findCellModuleName(cell: vscode.NotebookCell): string | undefined {
+  const cellText = cell.document.getText();
+  // Get all characters from cellText up until the \n character
+  // e.g. in the cells with the `#|export` only comment, the cell
+  // text will be `#|export\ndef get_id...` and we want to get `#|export` part only
+  // otherwise the regex matches `def get_id` which is wrong.
+  // This could be fixed with a better regex...
+  const textBeforeNewline = cellText.split("\n")[0];
+  const exportMatch = textBeforeNewline.match(/#\|export\s+(?!\n)(\w+)/);
+  if (exportMatch) {
+    const [_, name] = exportMatch;
+    return name;
+  }
+  return undefined;
+}
+
+export async function goToCustomDefinition() {
+  if (vscode.window.activeNotebookEditor) {
+    // We're in a notebook, so we're navigating back to code
+
+    // Find the cell that we're in
+    const notebookEditor = vscode.window.activeNotebookEditor;
+
+    const cell = vscode.window.activeNotebookEditor.notebook.cellAt(
+      notebookEditor.selections[0].start
+    );
+    if (!cell) {
+      return;
+    }
+
+    let moduleName = findCellModuleName(cell);
+
+    // We don't have a cell-local export, find the default one
+    if (!moduleName) {
+      // Get text from all cells in the document
+      const allCellsText = vscode.window.activeNotebookEditor.notebook
+        .getCells()
+        .map((cell) => cell.document.getText())
+        .join("");
+
+      // Search for the `#| default_exp` comment
+      const defaultExpMatch = allCellsText.match(/#\|\s*default_exp\s+(\w+)/);
+
+      if (defaultExpMatch) {
+        // Read the path from the default_exp comment
+        const [_, name] = defaultExpMatch;
+        moduleName = name;
+      }
+    }
+
+    // If the module name is still undefined, we can't do anything
+    // This happens if the user is in a cell that doesn't have an export comment
+    // and there is no default_exp comment in the document
+    if (!moduleName) {
+      return;
+    }
+
+    // Get the root of the workspace
+    const workspaceRoot = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+    if (!workspaceRoot) {
+      return;
+    }
+
+    // Open the settings.ini file
+    const settingsPath = path.join(workspaceRoot, "settings.ini");
```

Comment:
/hamel - think about making this more robust?  recurse "upwards" to find `settings.ini` like nbdev does?

Reply:
Actually move the settings.ini stuff to a utils thingy too